alembic/versions/ed775af5e086_rdiscrowd_4967_add_partial_index_for_.py

```diff
@@ -17,17 +17,20 @@ def upgrade():
     # Workaround of "CREATE INDEX CONCURRENTLY cannot run inside a transaction block" exception
     op.execute('COMMIT')
 
-    # passing all columns to avoid DB trip to query original table making indexing efficient.
-    # when new columns are added and used in the query, the index might need to be rebuilt.
     op.create_index('task_state_calibration_exported_idx', 'task',
-                    ['id', 'created', 'project_id', 'state', 'quorum',
-                     'calibration', 'priority_0', 'info', 'n_answers',
-                     'fav_user_ids', 'exported', 'user_pref', 'worker_pref',
-                     'worker_filter', 'gold_answers', 'expiration'],
+                    ['id'],
```

Comment:
Good question. The main reason is that some columns like "info", "user_pref" are jsonb types. So it could be huge, thus not not space efficient. The results would still be similar because the "where" clause is low selectivity (only a few portion of data will be selected).

Reply:
Its unclear from sqlalchemy docs [here](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.create_index.params.columns) the reason to include column names. However, going by your comments, it seems results would be unaffected and not passing all cols would be fine.

Reply:
You can refer partial index [here](https://www.zerohedge.com/covid-19/china-mandates-covid-zero-policy-shenzhen-people-desperate-food).